### ISSUE TITLE
Safe Mode: Restrict DELETE statements to cell/row/partition

### ISSUE THREAD

**Issue by slivne** [scylladb/scylla#9030]
DELETE operations can:
- delete a full partition
- delete a row
- delete a cell
- delete a range of rows (bounded & unbounded ranges) - aka range tombstones

We are enhancing the support for range tombstones - but allow clusters to be setup restricting this operations from being made

**Comment by nyh** [comment]
I didn't understand the motivation to forbid range deletes - is it because this feature is somehow inefficient (why?) or implemented incorrectly? Is there another open issue explaining the problem with range deletes?

**Comment by slivne** [comment]
Yes currently we have some inefficiency related to them  https://github.com/scylladb/scylla/issues?q=is%3Aopen+is%3Aissue+label%3A%22range+deletes%22

Please note that the restriction should be on the frontend disallowing this for the user not for scylla.

**Comment by eliransin** [comment]
@slivne just keep in mind that every such restriction have to be one that we know it's default value and stand behind it as it is going to be the default for every production system out there. 
What should we restrict here by default? If the default for this is off it can be implemented but I don't think it should be part of the
developer mode options.

**Comment by avikivity** [comment]
I think it's too restrictive. We'll just push people to use developer mode in production.

**Comment by eliransin** [comment]
> Yes currently we have some inefficiency related to them https://github.com/scylladb/scylla/issues?q=is%3Aopen+is%3Aissue+label%3A%22range+deletes%22

Warning against inefficiencies (in relation to the current implementation of Scylla) sounds like an `advisor` territory to me.

**Comment by tzach** [comment]
@avikivity can we close this?
